src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -110,12 +111,12 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 		return connection;
 	}
 
-	public $connect(connectionProfile: IConnectionProfile): Thenable<azdata.ConnectionResult> {
+	public $connect(connectionProfile: IConnectionProfile, saveConnection: boolean = false, showDashboard: boolean = false): Thenable<azdata.ConnectionResult> {
```

Comment:
We should default these to true so that existing callers don't have their behavior changed

Reply:
I don't think we need default values for these IPC methods.  They should only ever have 1 or 2 call points so I think explicitly listing parameters in those cases makes sense.  Default parameters are for methods that are used a lot and many parameters are often set a particular way.